rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+        also advertise information about operations that are not implemented
+        so that the catalog can eagerly throw or go about another way of performing
+        the desired action.
+      responses:
+        default:
+          description: Server-Specific Configuration Overrides
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergConfiguration'
+        "400":
+          description: Unknown Error
+        "401":
+          description: Unauthorized
+
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      - name: namespace
+        in: path
+        description: Namespace the table is in
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+      - name: table
+        in: path
+        description: Name of the table to load
+        required: true
+        schema:
+          type: string
+        example: "sales"
+    get:
+      tags:
+        - Catalog API
+      summary: Load a given table from a given namespace
+      operationId: loadTable
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetTableResponse'
+        '401':
+          description: Unauthorized
+        # Using 412, `Precondition Failed`, instead of 404, as 404 makes monitoring response codes from ELBs
+        #  very difficult - Hard to tell if clients or servers are misconfigured and calling non-existent routes
+        #  or missing routes versus expected error cases such as NoSuchTableException (expected meaning that
+        #  a person who is on call shouldn't be paged for this but 404 they might need to be).
+        '412':
+          description: NoSuchTableException
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchTableError'
+    put:
+      tags:
+        - Catalog API
+      summary: Commit an in progress create (or replace) table transaction
+      operationId: commitTable
+      description: Commit a pending create (or replace) table transaction, e.g. for doCommit.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CommitTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CommitTableResponse'
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergResponseObject'
+    delete:
```

Comment:
It seems that there is currently consensus around not tackling things that create a Transaction.

To make it easier to review this in parts, I'll remove anything that returns `Transaction` from this PR. That's these functions in the `Catalog` interface: https://github.com/apache/iceberg/blob/7bfecd8514b943b00415fc4f49d1c7626fe62f4c/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L129-L286

Dropping a table isn't technically a `Transaction`, but I do understand what you're saying. Dropping is a large table operation definitely.

I'm open to moving parts alongside the `Transaction` API into their own PR as well to make the initial discussion easier.

Reply:
Since `dropTable` isn't part of the transaction API, I'm going to leave it here for now at least. https://github.com/apache/iceberg/blob/27642de089aa1f9844b927fc700b29e299d7ca81/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L313-L322

If other's feel strongly about it, I'm open to that discussion. But at least from the API point of view, it doesn't have the same complications that `Transaction` based APIs have. And it also isn't part of `TableOperations`.

The goal isn't to necessarily prescribe how the server-side handles this, just give them a place to call the API.

They could return `202 - Accepted` or something as well and have it be somewhat async. And existing catalogs are free to still implement `dropTable` in the way they presently do.

I think a discussion of moving it to `TableOperations` should maybe be a separate discussion (if we have it).